File: app/src/test/java/igrek/songbook/dagger/base/TestModule.java
package igrek.songbook.dagger.base;

import android.support.v7.app.AppCompatActivity;

import igrek.songbook.dagger.FactoryModule;
import igrek.songbook.info.logger.Logger;
import igrek.songbook.mock.LoggerMock;

public class TestModule extends FactoryModule {
	
	public TestModule(AppCompatActivity activity) {
		super(activity);
	}
	
	@Override
	protected Logger provideLogger() {
		return new LoggerMock();
	}
	
}

File: app/src/test/java/igrek/songbook/dagger/base/TestModule.kt
package igrek.songbook.dagger.base

import androidx.appcompat.app.AppCompatActivity

import igrek.songbook.dagger.FactoryModule
import igrek.songbook.info.logger.Logger
import igrek.songbook.mock.LoggerMock

class TestModule(activity: AppCompatActivity) : FactoryModule(activity) {

    override fun provideLogger(): Logger {
        return LoggerMock()
    }

}
